core/middlewares/with-routes.ts

```diff
@@ -71,6 +71,8 @@ let locale: string;
 const updateRouteCache = async (pathname: string, event: NextFetchEvent): Promise<RouteCache> => {
   const channelId = getChannelIdFromLocale(locale);
 
+  kv.setChannelId(channelId);
```

Comment:
Would this be a problem if the code runs on the same isolate between requests for different channels? 🤔